File: longest_common_prefix.py
class Solution:
    def longestCommonPrefix(self, strs: list[str]) -> str:
        if not strs:
            return ""
        shortest = min(strs, key=len)
        for i, ch in enumerate(shortest):
            for other in strs:
                if other[i] != ch:
                    return shortest[:i]
        return shortest


s = Solution()
print(s.longestCommonPrefix(["dog","dogar","dogs"]))

File: golang/longest_common_prefix.go
package main

func longestCommonPrefix(strs []string) string {
	minLen := len(strs[0])
	for _, str := range strs {
		minLen = min(minLen, len(str))
	}

	result := ""
	for i := 0; i < minLen; i++ {
		current := strs[0][i]
		for _, str := range strs {
			if str[i] != current {
				return result
			}
		}

		result += string(current)
	}

	return result
}
